README.md

```diff
@@ -149,22 +149,29 @@ Book.bulk_insert(*destination_columns, ignore: true) do |worker|
 end
 ```
 
-### Update Duplicates (MySQL)
+### Update Duplicates (MySQL, PostgreSQL)
 
 If you don't want to ignore duplicate rows but instead want to update them
 then you can use the _update_duplicates_ option. Set this option to true
-and when a duplicate row is found the row will be updated with your new
-values. Default value for this option is false.
+(MySQL) or list unique column names (PostgreSQL) and when a duplicate row
+is found the row will be updated with your new values.
+Default value for this option is false.
 
 ```ruby
 destination_columns = [:title, :author]
 
-# Update duplicate rows
+# Update duplicate rows (MySQL)
 Book.bulk_insert(*destination_columns, update_duplicates: true) do |worker|
   worker.add(...)
   worker.add(...)
   # ...
 end
+
+# Update duplicate rows (PostgreSQL)
+Book.bulk_insert(*destination_columns, update_duplicates: %w[title]) do |worker|
```

Comment:
It may be a little bit misleading this change of argument type according to the adapter.

- I would support the option `update_duplicates:` boolean argument to be consistent in the gem implementation (this is the reason the of original ON CONFLICT DO NOTHING to reproduce the behaviour of INSERT_IGNORE https://github.com/jamis/bulk_insert/pull/19)

The options to pass to this argument could be:

- the columns to update (supported also in mysql https://dev.mysql.com/doc/refman/8.0/en/insert-on-duplicate.html)
- the conflicting columns (by default the primary key for MySQL)

Assuming that some has set correctly its primary keys, I would consider more intuitive that `update_duplicates:` will provide the list of the columns to update. This will make the option supported for MySQL as well.

Reply:
>  I would consider more intuitive that `update_duplicates:` will provide the list of the columns to update.

I would normally consider it a more intuitive as well, but Postgres is _requires inference specification or constraint name_ . I've spent a while on it now and I find it hard to come up with satisfying abstraction that will work both for MySQL and Postgres.

Reply:
what about considering the following scenarios:
- `false`, default
- `%[]`, only selected columns
- `true`, all destination columns
?

Reply:
I don't think it solves the issue. We have destination columns (data you insert) and we have update_duplicates (unique columns for postgres, which is done automatically by mysql).

> * `%[]`, only selected columns

How it would for MySQL?

> * `true`, all destination columns

How it would work for PostgreSQL?

Reply:
hey @sobstel , thank you for your patience 😄 

What I meant is the following:

Let's imagine we have a `foos` table with  columns `a`, `b`, `c`, `d`.

```rb
destination_columns = [:a, :b]
# I would see 3 use cases

# PG
Foo.bulk_insert(*destination_columns, update_duplicates: false) #clear
Foo.bulk_insert(*destination_columns, update_duplicates: [:a]) #clear
Foo.bulk_insert(*destination_columns, update_duplicates: true)
# => true will be a shortcut for update_duplicates: destination_columns

# Mysql
Foo.bulk_insert(*destination_columns, update_duplicates: false) #clear
Foo.bulk_insert(*destination_columns, update_duplicates: true) # clear
Foo.bulk_insert(*destination_columns, update_duplicates: [:a])
# => you will update only a
```

https://github.com/jamis/bulk_insert/blob/91b1a5b55318b8981b924193fbbcfc011ef7953a/lib/bulk_insert/worker.rb#L155-L159

New implementation suggested:

```rb
 elsif adapter_name =~ /^mysql/i && update_duplicates 
    columns_scope = \
       if  update_duplicates.is_a?Array
         ->(c) { update_duplicates.map(&:to_s).include?(c.name) }
       else
          ->(c) { c }
        end
   update_values = @columns.select(&columns_scope).map do |column| 
     "`#{column.name}`=VALUES(`#{column.name}`)" 
   end.join(', ') 
   ' ON DUPLICATE KEY UPDATE ' + update_values 
```

You can try it out in irb easily:

```
$ irb
irb(main):001:0> update_duplicates = ['a', :b, 1]
=> ["a", :b, 1]
irb(main):002:0> columns = ["b", "1"]
=> ["b", "1"]
irb(main):003:0> columns_scope = ->(c) { update_duplicates.map(&:to_s).include?(c) }
=> #<Proc:0x00007fa17980ba20@(irb):3 (lambda)>
irb(main):004:0> columns.select(&columns_scope)
=> ["b", "1"]
irb(main):005:0> id = ->(c) { c }
=> #<Proc:0x00007fa1768544d0@(irb):5 (lambda)>
irb(main):006:0> update_duplicates.select(&id)
=> ["a", :b, 1]
```

Reply:
Thanks your reply and sorry for another long wait. Unfortunately I struggle to find more time to invest into it, sorry. 

Actually you have implementation done above. Feel free to update it or even to create separate PR with final solution including your improvements.

Reply:
Would love to see this goes into code, we cannot use this amazing gem because lack of postgress support